src/jit/lower.cpp

```diff
@@ -2862,8 +2862,8 @@ void Lowering::InsertPInvokeMethodProlog()
     firstBlockRange.InsertBefore(insertionPoint, LIR::SeqTree(comp, store));
     DISPTREERANGE(firstBlockRange, store);
 
-#ifndef _TARGET_X86_ // For x86, this step is done at the call site (due to stack pointer not being static in the
-                     // function).
+#if !defined(_TARGET_X86_) && !defined(_TARGET_ARM_)
+    // For x86 and arm32, this step is done at the call site (due to stack pointer not being static in the function).
 
```

Comment:
> It seems that the approach to make ryujit output identical to the legacy one is not the best for every case.

That may be true for many things, but for many things, especially p/invoke prolog/epilog and call prolog/epilog, the code is so stylized that it is useful to either make it the same or understand the differences.

---
In reply to: [103537865](https://github.com/dotnet/coreclr/pull/9825#discussion_r103537865) [](ancestors = 103537865,103531744)